src/platform/nrfconnect/wifi/WiFiManager.cpp

```diff
@@ -564,5 +564,34 @@ void WiFiManager::AbortConnectionRecovery()
     Instance().mRecoveryTimerAborted = true;
 }
 
+CHIP_ERROR WiFiManager::SetLowPowerMode(bool onoff)
+{
+    net_if * iface = InetUtils::GetInterface();
+    VerifyOrReturnError(nullptr != iface, CHIP_ERROR_INTERNAL);
+
+    wifi_ps_config currentConfig{};
+    if (net_mgmt(NET_REQUEST_WIFI_PS_CONFIG, iface, &currentConfig, sizeof(currentConfig)))
+    {
+        ChipLogError(DeviceLayer, "Get current low power mode config request failed");
+        return CHIP_ERROR_INTERNAL;
+    }
+
+    if ((currentConfig.enabled == WIFI_PS_ENABLED && onoff == false) ||
+        (currentConfig.enabled == WIFI_PS_DISABLED && onoff == true))
+    {
+        wifi_ps_params params{ .enabled = onoff ? WIFI_PS_ENABLED : WIFI_PS_DISABLED };
```

Comment:
Would it make sense to modify `currentConfig` instead of creating a new structure, to preserve other PS settings? Like whether it's WMM or legacy.

Reply:
Not sure if I get this right but if you mean reusing the currentConfig in the `net_mgmt(NET_REQUEST_WIFI_PS,...)` call it's a different API: `wifi_ps_config` vs. `wifi_ps_params`. But we can preserve the current power save mode configuration for the future purposes, i.e. by by extending the ConnectionParams structure with `wifi_ps_config` field.

Reply:
ah, right. I thought it's a single structure used for both APIs. Nevermind then.